ceph_deploy/hosts/remotes.py

```diff
@@ -158,26 +158,25 @@ def remove_whitespace_from_assignments():
 def write_conf(cluster, conf, overwrite):
     """ write cluster configuration to /etc/ceph/{cluster}.conf """
     path = '/etc/ceph/{cluster}.conf'.format(cluster=cluster)
-    tmp_file = tempfile.NamedTemporaryFile('w', dir='/etc/ceph', delete=False)
-    err_msg = 'config file %s exists with different content; use --overwrite-conf to overwrite' % path
+    # Stateless OS like ClearLinux does not have anything populated in /etc.
+    # Create the folder before we write the config file.
+    if not os.path.isdir('/etc/ceph'):
```

Comment:
@ganeshmaharaj this changes the existing behavior even on non-ClearLinux distros.

before your change, we bail out if `/etc/ceph` does not exist with the error message of 

> '/etc/ceph/ does not exist - could not write config'

but after your change this directory is always created as long as it does not exist beforehand. i understand Clear Linux tries to avoid placing anything in `/etc`, see https://clearlinux.org/news-blogs/where-etcfstab-clear-linux . does this imply that `swupd` does not create `/etc/ceph` for us when/after installing ceph packages?

if that's the intended behavior of `swupd` and the ceph packages on ClearLinux, how about creating `/etc/ceph` in `Swupd.install()` ? so we don't change the existing behavior on other distros?

Reply:
@tchaikov yes, clearlinux does not create `/etc/ceph` when we install packages by default. I will move the folder creation to Swupd.install() and get the changes here back to where they were.